src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentColor/DocumentColorEndpoint.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.DocumentColor
+{
+    internal class DocumentColorEndpoint : IDocumentColorHandler
+    {
+        private static readonly Container<ColorInformation> EmptyDocumentColors = new Container<ColorInformation>();
+        private readonly ClientNotifierServiceBase _languageServer;
+
+        public DocumentColorEndpoint(ClientNotifierServiceBase languageServer)
+        {
+            if (languageServer is null)
+            {
+                throw new ArgumentNullException(nameof(languageServer));
+            }
+
+            _languageServer = languageServer;
+        }
+        public DocumentColorRegistrationOptions GetRegistrationOptions(ColorProviderCapability capability, ClientCapabilities clientCapabilities)
+        {
+            return new DocumentColorRegistrationOptions()
+            {
+                DocumentSelector = RazorDefaults.Selector,
+            };
+        }
+
+        public async Task<Container<ColorInformation>> Handle(DocumentColorParams request, CancellationToken cancellationToken)
+        {
+            var delegatedRequest = await _languageServer.SendRequestAsync(LanguageServerConstants.RazorProvideHtmlDocumentColorEndpoint, request).ConfigureAwait(false);
+            var documentColors = await delegatedRequest.Returning<Container<ColorInformation>?>(cancellationToken).ConfigureAwait(false);
+
+            if (documentColors is null)
+            {
+                return EmptyDocumentColors;
```

Comment:
Why can't we just return null here?

Reply:
Spec doesn't support `null` supposedly

Reply:
Seems like WebTools does 😛

Reply:
😄 lol fair, we're being a little protective here aren't we

Reply:
To be extra defensive for now I'll leave this as is given it shouldn't harm anything.